checkov/common/output/ai.py

```diff
@@ -29,10 +30,32 @@
 class OpenAi:
     _instance = None  # noqa: CCE003  # singleton
 
-    def __new__(cls, api_key: str | None = None) -> Self:
+    def _validate_azure_env(self, value: str | None = None, environment_variable_name: str | None = None) -> bool:
+        if (value is None):
+            print(
+                colored(
+                    f"ERROR: Configuration for Azure OpenAI is missing: Please specify {environment_variable_name} environment variable for --openai-api-type '{self._api_type}' type.",
+                    "red",
+                )
+            )
+            return False
+        return True
+
+    def __new__(cls, api_key: str | None = None, api_type: str = "default") -> Self:
         if cls._instance is None:
             cls._instance = super().__new__(cls)
             cls._should_run = True if api_key else False
+            cls._api_type = api_type.lower()
+            if (cls._api_type == 'azure'):
+                cls.AZURE_OPENAI_API_ENDPOINT = os.getenv("CKV_AZURE_OPENAI_API_ENDPOINT", None)
+                cls.AZURE_OPENAI_API_VERSION = os.getenv("CKV_AZURE_OPENAI_API_VERSION", '2023-05-15')
+                cls.AZURE_OPENAI_DEPLOYMENT_NAME = os.getenv("CKV_AZURE_OPENAI_DEPLOYMENT_NAME", None)
+                cls._should_run = cls._should_run & cls._validate_azure_env(cls._instance, cls.AZURE_OPENAI_API_ENDPOINT, 'CKV_AZURE_OPENAI_API_ENDPOINT')
+                cls._should_run = cls._should_run & cls._validate_azure_env(cls._instance, cls.AZURE_OPENAI_API_VERSION, 'CKV_AZURE_OPENAI_API_VERSION')
+                cls._should_run = cls._should_run & cls._validate_azure_env(cls._instance, cls.AZURE_OPENAI_DEPLOYMENT_NAME, 'CKV_AZURE_OPENAI_DEPLOYMENT_NAME')
```

Comment:
instead of validating each one by one, pass all to a validate function and decide, if everything is set as needed